src/main/java/spoon/support/reflect/cu/CompilationUnitImpl.java

```diff
@@ -42,7 +42,7 @@
 
 	CtPackage ctPackage;
 
-	Collection<CtReference> imports = new HashSet<>();
+	Collection<CtImport> imports = new HashSet<>();
```

Comment:
What about using List instead of Set... here and in JDTImportBuilder#imports too, then we have a chance to keep the origin order of imports - minimize changes in generated sources.

Reply:
> here and in JDTImportBuilder#imports too, then we have a chance to keep the origin order of imports - minimize changes in generated sources.

I agree, but I propose to do it in another PR: this one start to be difficult to understand. So I propose that you open an issue for that, and I'll make new tests and changes to assert that. WDYT?